nengo/connection.py

```diff
@@ -91,9 +98,9 @@ def check_transform(self, conn, rule):
 
         if rule.modifies == "weights":
             # If the rule modifies 'weights', then it must have full weights
-            if conn.is_decoded:
+            if not conn._to_neurons:
```

Comment:
I'm not convinced that all "weights" learning rules will work for all `_to_neurons` connections. We should test this more thoroughly, but it probably makes sense to do that with the NEFTransform changes.

Reply:
Yeah I suspect that lots of other learning rules will have similar issues to the bugs fixed here, I just didn't want to get into checking/debugging them all right now (and figured it wasn't as urgent since they're less commonly used).